### PR TITLE
Add Using ref with React Component to Docs

### PR DESCRIPTION
**Pull request by stream7**
Adds a documentation page regarding React's `ref`

### CODE REVIEW

File: docs/recipes/UsingRefWithReactComponent.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+const StyledInput = createComponent(rules, 'input')
+
+class Form extends Component {
+  componentDidMount() {
+    this.input.focus()
+  }
+
+  render() {
+    return (
+      <StyledInput innerRef={ref => { this.input = comp }} />
```
Comment (tajo) [inline docs/recipes/UsingRefWithReactComponent.md]:
```jsx
<StyledInput innerRef={ref => { this.input = ref }} />
```

?

Comment (stream7) [inline docs/recipes/UsingRefWithReactComponent.md]: @tajo not sure I understand the question 😸

Comment (stream7) [inline docs/recipes/UsingRefWithReactComponent.md]: oops, got it, fixing

Comment (stream7) [inline docs/recipes/UsingRefWithReactComponent.md]: Updated

**Comment by tajo** [comment]
Cool, thanks!

**Pull request merged**